src/test/scala/org/broadinstitute/dsde/firecloud/mock/MockTSV.scala

```diff
@@ -316,6 +316,13 @@ object MockTSVLoadFiles {
   val missingFields1 = TSVLoadFile("foo", Seq("foo", "bar", "baz"), Seq(Seq("biz", "", "buz")))
   val missingFields2 = TSVLoadFile("foo", Seq("foo", "bar", "baz"), Seq(Seq("", "", "buz"), Seq("abc", "123", "")))
 
+  val entityWithAttributeBooleanArray = TSVLoadFile("array", Seq("array"), Seq(Seq("bla", """[false,true,true]""")))
+  val entityWithAttributeNumberArray = TSVLoadFile("array", Seq("array"), Seq(Seq("bla", """[1,2,3]""")))
+  val entityWithAttributeStringArray = TSVLoadFile("array", Seq("array"), Seq(Seq("bla", """["foo","bar","baz"]""")))
+  val entityWithAttributeMixedArray = TSVLoadFile("array", Seq("array"), Seq(Seq("bla", """[false,"foo",1]""")))
+  val entityWithAttributeArrayOfObjects = TSVLoadFile("array", Seq("array"), Seq(Seq("bla", """[{"one":"two"},{"three":"four"},{"five":"six"}]""")))
+  val entityWithEmptyAttributeArray = TSVLoadFile("array", Seq("array"), Seq(Seq("bla", """[]""")))
```

Comment:
Should there also be a mock here for malformed json?

Reply:
And a unit test for unit as well

Reply:
@ungwudik what do you mean by "unit test for unit"?

I don't think it's possible to cleanly distinguish a malformed json array. Your typical TSV will contain mostly strings and numbers … all of which technically do not parse as json arrays. So, if your TSV contains `"hello world"` and `123`, both of those cells will fail to parse as arrays. We can't warn or alert the user on those, because they're totally valid, they're just not json arrays. Inside the Scala logic, we can't really distinguish between `"hello world"` and `["hello", "world">` … there's no indicator for "close to" a json array.

Reply:
Sorry, typo. I meant "And a unit test for **it(malformed json array)** as well"